solr/core/src/java/org/apache/solr/handler/component/QueryElevationComponent.java

```diff
@@ -682,35 +686,48 @@ public String getDescription() {
    * Creates the {@link ElevationProvider} to set during configuration loading. The same instance will be used later
    * when elevating results for queries.
    *
-   * @param queryAnalyzer to analyze and tokenize the query.
    * @param elevationBuilderMap map of all {@link ElevatingQuery} and their corresponding {@link ElevationBuilder}.
    * @return The created {@link ElevationProvider}.
    */
-  protected ElevationProvider createElevationProvider(Analyzer queryAnalyzer, Map<ElevatingQuery, ElevationBuilder> elevationBuilderMap) {
-    return new MapElevationProvider(elevationBuilderMap);
+  protected ElevationProvider createElevationProvider(Map<ElevatingQuery, ElevationBuilder> elevationBuilderMap) {
+    return new SubsetMatchElevationProvider(new TrieSubsetMatcher.Builder<>(), elevationBuilderMap);
   }
 
   //---------------------------------------------------------------------------------
   // Query analysis and tokenization
   //---------------------------------------------------------------------------------
 
   /**
-   * Analyzes the provided query string and returns a space concatenation of the analyzed tokens.
+   * Analyzes the provided query string and returns a concatenation of the analyzed tokens.
    */
   public String analyzeQuery(String query) {
-    //split query terms with analyzer then join
-    StringBuilder norm = new StringBuilder();
+    StringBuilder concatenatedTerms = new StringBuilder();
+    analyzeQuery(query, null, concatenatedTerms);
+    return concatenatedTerms.toString();
+  }
+
+  /**
+   * Analyzes the provided query string, tokenizes the terms and add them to either the provided {@link Collection} or {@link Appendable}.
+   *
+   * @param queryTerms The {@link Collection} that receives the terms; or null if none.
+   * @param concatenatedTerms The {@link Appendable} that receives the terms; or null if none.
+   */
+  protected void analyzeQuery(String query, Collection<String> queryTerms, Appendable concatenatedTerms) {
```

Comment:
This method's contract is a little awkward in that it has dual optional output methods, but it's okay.  Maybe it's be nicer if it took a Consumer<CharSequence> since the set addition and appending are trivial for the caller to do?  Up to you; just an idea.

Reply:
I pondered for a while on this signature, because I wanted it to be clear but also performant where it is used, to avoid the creation of lots of lambdas in loops.
I'll change back to the Consumer. It will create just an additional lambda instance per query to process, given that I'll declare the lambdas out of the loops.